src/test/java/io/strimzi/kafka/metrics/YammerMetricsCollectorTest.java

```diff
@@ -48,35 +53,74 @@ public void testCollect() {
         metrics = collector.collect();
         assertTrue(metrics.isEmpty());
 
-        // Adding a non-numeric metric does nothing
-        newNonNumericGauge("group", "name2", "type");
-        metrics = collector.collect();
-        assertTrue(metrics.isEmpty());
-
         // Adding a metric that matches the allowlist
         Counter counter = newCounter("group", "name", "type");
         metrics = collector.collect();
         assertEquals(1, metrics.size());
+
+        Collector.MetricFamilySamples metricFamilySamples = metrics.get(0);
+        Collector.MetricFamilySamples.Sample serverGroupNameSamples = metricFamilySamples.samples.get(0);
+
         assertEquals("kafka_server_group_name_type_count", metrics.get(0).name);
-        assertEquals(1, metrics.get(0).samples.size());
-        assertEquals(0.0, metrics.get(0).samples.get(0).value, 0.1);
-        assertEquals(new ArrayList<>(tags.keySet()), metrics.get(0).samples.get(0).labelNames);
-        assertEquals(new ArrayList<>(tags.values()), metrics.get(0).samples.get(0).labelValues);
+        assertEquals(1, metricFamilySamples.samples.size());
+        assertEquals(0.0, serverGroupNameSamples.value, 0.1);
+        assertEquals(new ArrayList<>(tags.keySet()), serverGroupNameSamples.labelNames);
+        assertEquals(new ArrayList<>(tags.values()), serverGroupNameSamples.labelValues);
 
         // Updating the value of the metric
         counter.inc(10);
         metrics = collector.collect();
         assertEquals(1, metrics.size());
-        assertEquals("kafka_server_group_name_type_count", metrics.get(0).name);
-        assertEquals(1, metrics.get(0).samples.size());
-        assertEquals(10.0, metrics.get(0).samples.get(0).value, 0.1);
+
+        Collector.MetricFamilySamples metricFamilySamples1 = metrics.get(0);
```

Comment:
No need to introduce new variables here, you can reassign the ones we created above.